### PR TITLE
Make ActiveStorage::Filename#sanitized replacement character configurable

### PR DESCRIPTION
**Pull request by LaurensN**
### Summary

Ability to configure the character that is used for replacing invalid characters in `ActiveStorage::Filename#sanitized`

Users migrating paperclip to activestorage have probably used `Paperclip::Attachment.cleanup_filename` that used an underscore by default.

By introducing this configurable option the migration is a lot easier (or makes it possible to just keep the behaviour of paperclip).

### CODE REVIEW

**Comment by kaspth** [comment]
Dubious about adding this solely for migration purposes directly in Rails. Could tr(“-“, “_”) suitably be called on the sanitized return value in your app?

Alternatively have you considered making an activestorage paper clip migration gem?

**Comment by LaurensN** [comment]
Hello @kaspth,

A simple `tr(“-“, “_”)` wouldn't suffice sadly:

```
[2] pry(main)> Paperclip::Attachment.cleanup_filename "../Filename-January-123$.png"
=> ".._Filename-January-123_.png"

[3] pry(main)> ActiveStorage::Filename.new("../Filename-January-123$.png").sanitized.tr(“-“, “_”)
=> ".._Filename_January_123_.png"
```

We want to be able to determine the character that is used for sanitization, not just for the migration to active_storage. We could ofcourse monkeypatch `ActiveStorage::Filename#sanitized`, but I figured that there maybe would be more people with the same wish, hence this PR. 

A migration gem would also require to update all the old attachments, this is not feasible for us.

**Comment by kaspth** [comment]
> We want to be able to determine the character that is used for sanitization, not just for the migration to active_storage.

I thought the reason to control sanitization was to migrate from Paper Clip. What's the actual use case for this?

> We could ofcourse monkeypatch `ActiveStorage::Filename#sanitized`, but I figured that there maybe would be more people with the same wish, hence this PR.

Based on what I read it all sounded like this was purely for migrating from Paper Clip to Active Storage. And in that case it would be way better for everyone if someone took the lead and wanted to maintain a monkey patch and whatever else for turn-key migration to Active Storage.

**Comment by LaurensN** [comment]
>I thought the reason to control sanitization was to migrate from Paper Clip. What's the actual use case for this?

For us the use case is transitioning all our filename sanitization to `ActiveStorage::Filename#sanitized`. We also sanitize filenames that are not directly added via paperclip. By introducing this configurable setting we are able to just keep the old paperclip behavior, this way we can use the `ActiveStorage` sanitizer and remove the Paperclip dependency.

We'd rather just use active_storage with this setting instead of using a monkeypatch. A one time migration is also hard because it is almost impossible to tell which underscores (in our case) are the result of sanitization, and which ones are supposed to be in the filename.

So basically the use case is just keeping the paperclip behavior, the best/easiest solution in our case.

**Comment by kaspth** [comment]
Cool, got it. I'm not convinced that we should add support for this, but I'm not too unconvinced that I want to close it outright. I'll keep this open to see if anyone else is swayed. Maybe @georgeclaghorn has thoughts?
